lib/internal/util.js

```diff
@@ -275,24 +272,25 @@ function promisify(original) {
   const argumentNames = original[kCustomPromisifyArgsSymbol];
 
   function fn(...args) {
-    const promise = createPromise();
-    try {
-      original.call(this, ...args, (err, ...values) => {
-        if (err) {
-          promiseReject(promise, err);
-        } else if (argumentNames !== undefined && values.length > 1) {
-          const obj = {};
-          for (var i = 0; i < argumentNames.length; i++)
-            obj[argumentNames[i]] = values[i];
-          promiseResolve(promise, obj);
-        } else {
-          promiseResolve(promise, values[0]);
-        }
-      });
-    } catch (err) {
-      promiseReject(promise, err);
-    }
-    return promise;
+    return new Promise((resolve, reject) => {
+      try {
```

Comment:
This `try/catch` is no longer necessary since the promise constructor converts `throw`s to rejects automatically.